src/firmware/types.rs

```diff
@@ -202,3 +208,522 @@ pub struct SnpPlatformStatus {
     /// Reported TCB version.
     pub reported_tcb_version: TcbVersion,
 }
+
+#[repr(C, packed)]
+pub struct SnpConfig {
+    reported_tcb: u64,
+    mask_chip_id: u32,
+    reserved: [u8; 52],
+}
+
+impl Default for SnpConfig {
+    fn default() -> Self {
+        Self {
+            reported_tcb: Default::default(),
+            mask_chip_id: Default::default(),
+            reserved: [0; 52],
+        }
+    }
+}
+
+#[repr(C)]
+pub struct SnpSetExtConfig<'a> {
+    /// Address of the SnpConfig or 0 when reported_tcb does not need
+    /// to be updated.
+    pub config_address: Option<&'a SnpConfig>,
+
+    /// Address of extended guest request certificate chain or 0 when
+    /// previous certificate should be removed on SNP_SET_EXT_CONFIG.
+    pub certs_address: Option<*mut c_void>,
+
+    /// Length of the certificates.
+    pub certs_len: u32,
+}
+
+#[repr(C)]
+pub struct SnpGetExtConfig<'a> {
+    /// Address of the SnpConfig or 0 when reported_tcb should not be
+    /// fetched.
+    pub config_address: Option<&'a SnpConfig>,
```

Comment:
Like my above comment, pointers are represented as `u64`, and Rust-friendly types would need to be built to interact with the APIs.

Reply:
I created a generic Rust friendly `SnpExtConfig` which will be returned from the ioctl functions once they are implemented.